app/controllers/concerns/cookie_concern.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module CookieConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def set_default_cookies
+    return set_default_analytics_cookies if no_analytics_cookies?
+
+    if user_set_analytics_preference?
+      set_cookie(:analytics_cookies_set, value: params[:analytics_cookies_set])
+      set_cookie(:cookies_preferences_set, value: true)
+    end
+
+    set_analytics_cookies
+    show_hide_cookie_banners
+  end
+
+  def set_cookie(type, value: false)
+    cookies[type] = {
+      value: value,
+      domain: request.host,
+      expires: Time.zone.now + 1.year
+    }
+  end
+
+  def set_default_analytics_cookies
+    set_cookie(:analytics_cookies_set)
+  end
+
+  def no_analytics_cookies?
+    cookies[:analytics_cookies_set].nil?
+  end
+
+  def user_set_analytics_preference?
+    params[:analytics_cookies_set].present?
+  end
+
+  def set_analytics_cookies
+    @analytics_cookies_accepted = ActiveModel::Type::Boolean.new.cast(cookies[:analytics_cookies_set])
+    return if @analytics_cookies_accepted
+
+    remove_analytics_cookies
+  end
+
+  def remove_analytics_cookies
```

Comment:
I would not delete these two generic GA cookies, as they may affect other services on the `.justice.gov.uk` domain that are using GA.

Reply:
So when the analytics cookies are rejected, the user won't be tracked if those GA cookies remain as long as the script itself is removed?

Reply:
If the user rejects cookies, we need to remove the cookies.  Perhaps this is an argument for having a more joined-up MoJ strategy, but for the moment, since these cookies are added when the user clicks "accept", they need to be removed when the user clicks "reject".

Reply:
Do we not need to delete the _ga, _gid cookies at all as long as the GA script is not being rendered?